regression/deterministic_shuffling.yaml

```diff
@@ -0,0 +1,42 @@
+name: streaming-regression-test-deterministic-shuffling
+compute:
+  gpus: 8 # Number of GPUs to use
+  # cluster: TODO # Name of the cluster to use for this run
+command: |-
+  pip uninstall -y mosaicml-streaming
+  cd streaming
+  pip install -e '.[dev]'
+
+  python regression/synthetic_dataset.py --create --out /tmp/streaming_dataset/
+
+  # Check sample order without shuffling
+  composer -n 2 regression/iterate_data.py --num_canonical_nodes 1 --batch_size 1 --local /tmp/local_dataset/ --remote /tmp/streaming_dataset/ --shuffling-validation-file /tmp/test_samples_0.txt
```

Comment:
Can we also verify `batch_size=16 and/or 32`  ? The `batch_size=1` does not touch some of data partition logic in streaming, putting it a higher value would ensure that those code part gets tested. You can even replace `batch_size` of `1` with `32` !

Reply:
Can you do the same with `num_canonical_nodes` ? Let's vary such as  `128` and `256` and default